zarr/storage.py

```diff
@@ -110,7 +115,159 @@ def init_store(store, shape, chunks, dtype=None, compression='default',
     store['attrs'] = json.dumps(dict()).encode('ascii')
 
 
-class DirectoryStore(MutableMapping):
+# backwards compatibility
+init_store = init_array
+
+
+def init_group(store, overwrite=False):
+    """Initialise a group store.
+
+    Parameters
+    ----------
+    store : MutableMapping
+        A mapping that supports string keys and byte sequence values.
+    overwrite : bool, optional
+        If True, erase all data in `store` prior to initialisation.
+
+    """
+
+    # delete any pre-existing items in store
+    if overwrite:
+        store.clear()
+
+    # initialise attributes
+    if 'attrs' not in store:
+        store['attrs'] = json.dumps(dict()).encode('ascii')
+
+
+def check_array(store):
+    return 'meta' in store
```

Comment:
Good point, this needs some thought. There needs to be some reliable convention for telling if a store contains a Zarr array or a Zarr group or neither.